src/main/webapp/app/exam/participate/exam-participation.service.ts

```diff
@@ -246,4 +246,49 @@ export class ExamParticipationService {
             });
         }
     }
+
+    public getSubmissionForExercise(exercise: Exercise) {
+        if (
+            exercise &&
+            exercise.studentParticipations &&
+            exercise.studentParticipations.length > 0 &&
+            exercise.studentParticipations[0].submissions &&
+            exercise.studentParticipations[0].submissions.length > 0
+        ) {
+            return exercise.studentParticipations[0].submissions[0];
+        } else {
+            return undefined;
+        }
+    }
+
+    getExerciseButtonTooltip(
+        exercise: Exercise,
+        submission = this.getSubmissionForExercise(exercise),
+    ): 'submitted' | 'notSubmitted' | 'synced' | 'notSynced' | 'notSavedOrSubmitted' {
+        console.log('exercise, submission', exercise, submission);
+        if (submission) {
+            if (exercise.type === ExerciseType.PROGRAMMING) {
+                if (submission.submitted && submission.isSynced) {
+                    return 'submitted'; // You have submitted an exercise. You can submit again
+                } else if (submission.submitted && !submission.isSynced) {
+                    return 'notSavedOrSubmitted'; // You have unsaved and/or unsubmitted changes
+                } else if (!submission.submitted && submission.isSynced) {
+                    return 'notSubmitted'; // starting point
+                } else {
+                    return 'notSavedOrSubmitted';
+                }
+            } else {
+                if (submission.isSynced) {
+                    return 'synced';
+                } else {
+                    return 'notSynced';
+                }
+            }
+        } else {
+            // submission does not yet exist for this exercise.
+            // When the participant navigates to the exercise the submissions are created.
+            // Until then show, that the exercise is synced
+            return 'synced';
+        }
+    }
```

Comment:
Imho this is much less complex as
```suggestion
        // submission does not yet exist for this exercise.
        // When the participant navigates to the exercise the submissions are created.
        // Until then show, that the exercise is synced
        if (!submission) {
            return 'synced';
        }

        if (exercise.type !== ExerciseType.PROGRAMMING) {
            return submision.isSynced ? 'synced' : 'notSynced';
        }

        if (submission.submitted && submission.isSynced) {
            return 'submitted'; // You have submitted an exercise. You can submit again
        } else if (submission.submitted && !submission.isSynced) {
            return 'notSavedOrSubmitted'; // You have unsaved and/or unsubmitted changes
        } else if (!submission.submitted && submission.isSynced) {
            return 'notSubmitted'; // starting point
        } else {
            return 'notSavedOrSubmitted';
        }
```

Reply:
great catch! Done